### ISSUE TITLE
Strapi gives 403 error when updating collection 🚨when using replica of database

### ISSUE THREAD

**Issue by sehrish30** [strapi/strapi#19704]
### Required System information

- Node.js version: 18
- NPM version: 9.8.1
- Strapi version:  4.19.1
- Database: Mysql
- Operating system: Ubuntu
- Is your project Javascript or Typescript: Typescript

### Describe the bug

When attempting to update data in a replicated database using Strapi, I encounter a 403 error specifically for a certain collection. 

1) if I create a new item in a collection and try to save or update it works fine.
2) If I duplicate the existing record in a collection, I'm still not able to update it it gives me 403 error, although I created it so I don't think it's related to permissions
3) I'm able to create a duplicate of records.
4) I'm able to delete a record
5) But I'm not able to update record.

### Steps to reproduce the behavior

1. Go to 'collection with big schema'
2. Try to update something
3. Click on save
4. Get 403 response "Forbidden"

### Expected behavior

Save data and show message "successfully saved"

### Screenshots

<img width="472" alt="image" src="https://github.com/strapi/strapi/assets/66527974/599a86f1-c98d-48f8-b98e-3d8c7576269b">

### Additional context

The permissions are same in my other project this is replica of that project still when I try to update that particular collection type I get 403 denied while I can save other collections. Note: this is the biggest collection in the project. If I create a new item in this collection I can save it and update it as well. 

This is clone of other project and database is cloned as well so permissions and data are eventually same. I get no error but the button always stays "Save"
<img width="235" alt="image" src="https://github.com/strapi/strapi/assets/66527974/519b6cc6-12c1-4821-bdaf-6360cc0348d5">

Because of VPN issues I'm not able to transfer data using strapi transfer token yet and the original project with this data works absolutely fine 🫰.

Can anyone please guide me how can I solve this issue? 💁

**Comment by paulhuisman** [comment]
Did you manage to solve this? I'm also getting random 403 errors on production when trying to save a single type

**Comment by sehrish30** [comment]
> Did you manage to solve this? I'm also getting random 403 errors on production when trying to save a single type

Yeah, turned out it was a permission issue with the domain-level security
